internal/transport/http2_client.go

```diff
@@ -346,7 +347,7 @@ func newHTTP2Client(connectCtx, ctx context.Context, addr resolver.Address, opts
 		streamQuota:           defaultMaxStreamsClient,
 		streamsQuotaAvailable: make(chan struct{}, 1),
 		keepaliveEnabled:      keepaliveEnabled,
-		bufferPool:            newBufferPool(),
+		bufferPool:            opts.BufferPool,
```

Comment:
Earlier, there was one shared pool per transport. Now, there is one shared pool per clientConn. And this is going to be used for the send and receive sides. Does this change anything, like increased contention while `getting` and `putting` buffers from and to the pool?

Reply:
Hmmm that's an interesting question. I think there should be some benefits to sharing as much as possible but it's possible it could indeed increase contention. Probably something worth benchmarking?